### ISSUE TITLE
Crash: -[SDImageIOCoder encodedDataWithImage:format:options:]

### ISSUE THREAD

**Issue by iwheelbuy** [SDWebImage/SDWebImage#2879]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 13.1.3
 SDWebImage Version      | 5.2.3
 Integration Method      | cocoapods
 Xcode Version           | Xcode 11
 Repro rate              | only once
 Repro with our demo prj | no
 Demo project link       | -

### Issue Description and Steps
```
SDImageIOCoder.m line 224
-[SDImageIOCoder encodedDataWithImage:format:options:]
```
```
Fatal Exception: NSMallocException
Failed to grow buffer
```
```
# Organization: PRJCT
# Platform: ios
# Application: PRJCT
# Bundle Identifier: ru.PRJCT.PRJCTapp
# Date: 2019-10-20T19:37:00Z
# OS Version: 13.1.3 (17A878)
# Device: iPhone X
# RAM Free: 3.4%
# Disk Free: 4.8%

#7. Crashed: com.twitter.crashlytics.ios.exception
0  PRJCT Prod                     0x104ea02a0 CLSProcessRecordAllThreads + 376 (CLSProcess.c:376)
1  PRJCT Prod                     0x104ea0688 CLSProcessRecordAllThreads + 407 (CLSProcess.c:407)
2  PRJCT Prod                     0x104e9038c CLSHandler + 26 (CLSHandler.m:26)
3  PRJCT Prod                     0x104e9e8b4 __CLSExceptionRecord_block_invoke + 198 (CLSException.mm:198)
4  libdispatch.dylib              0x1b169e184 _dispatch_client_callout + 16
5  libdispatch.dylib              0x1b1681dd8 _dispatch_lane_barrier_sync_invoke_and_complete + 56
6  PRJCT Prod                     0x104e9e35c CLSExceptionRecord + 205 (CLSException.mm:205)
7  PRJCT Prod                     0x104e9e190 CLSExceptionRecordNSException + 102 (CLSException.mm:102)
8  PRJCT Prod                     0x104e9ddb4 CLSTerminateHandler() + 258 (CLSException.mm:258)
9  libc++abi.dylib                0x1b179e304 std::__terminate(void (*)()) + 16
10 libc++abi.dylib                0x1b179dc58 __cxa_get_exception_ptr + 30
11 libc++abi.dylib                0x1b179dc18 __cxxabiv1::exception_cleanup_func(_Unwind_Reason_Code, _Unwind_Exception*) + 122
12 libobjc.A.dylib                0x1b16f91d0 _objc_exception_destructor(void*) + 354
13 CoreFoundation                 0x1b1a2c2b8 _CFPreferencesCopyManagedPreferencesContainerURL.cold.1 + 46
14 CoreFoundation                 0x1b1a24b54 __CFReallocationFailed + 120
15 CoreFoundation                 0x1b18ee0d0 __CFSafelyReallocate + 68
16 Foundation                     0x1b1d4efa4 _NSMutableDataGrowBytes + 344
17 Foundation                     0x1b1c9aaa8 -[NSConcreteMutableData appendBytes:length:] + 336
18 ImageIO                        0x1b22089fc IIOImageWriteSession::putBytes(void const*, unsigned long) + 184
19 ImageIO                        0x1b21df0e8 iioWriteCallback(void*, int, void*) + 36
20 AppleJPEG                      0x1dbab69ec aj_ostream_write + 392
21 AppleJPEG                      0x1dbad85dc aj_block_encode_ac + 1336
22 AppleJPEG                      0x1dbaa8990 aj_encode_buffers_baseline + 52
23 AppleJPEG                      0x1dbaa8dc4 aj_encode_row + 180
24 AppleJPEG                      0x1dbad9d84 applejpeg_encode_image_row + 580
25 AppleJPEG                      0x1dbac0d24 recode_all + 1300
26 ImageIO                        0x1b228c998 IIOImageDestination::finalize() + 5040
27 ImageIO                        0x1b2291158 CGImageDestinationFinalize + 124
28 PRJCT Prod                     0x104e1ad1c -[SDImageIOCoder encodedDataWithImage:format:options:] + 224 (SDImageIOCoder.m:224)
29 PRJCT Prod                     0x104e18130 -[SDImageCodersManager encodedDataWithImage:format:options:] + 127 (SDImageCodersManager.m:127)
30 PRJCT Prod                     0x104e10530 __71-[SDImageCache storeImage:imageData:forKey:toMemory:toDisk:completion:]_block_invoke + 197 (SDImageCache.m:197)
31 libdispatch.dylib              0x1b169d610 _dispatch_call_block_and_release + 24
32 libdispatch.dylib              0x1b169e184 _dispatch_client_callout + 16
33 libdispatch.dylib              0x1b167b73c _dispatch_lane_serial_drain$VARIANT$armv81 + 564
34 libdispatch.dylib              0x1b167c154 _dispatch_lane_invoke$VARIANT$armv81 + 400
35 libdispatch.dylib              0x1b168543c _dispatch_workloop_worker_thread + 576
36 libsystem_pthread.dylib        0x1b16edfa4 _pthread_wqthread + 276
37 libsystem_pthread.dylib        0x1b16f0ae0 start_wqthread + 8

--

Fatal Exception: NSMallocException
0  CoreFoundation                 0x1b19d098c __exceptionPreprocess
1  libobjc.A.dylib                0x1b16f90a4 objc_exception_throw
2  CoreFoundation                 0x1b1a2c2b8 _CFPreferencesCopyManagedPreferencesContainerURL.cold.1
3  CoreFoundation                 0x1b1a24b54 __CFReallocationFailed
4  CoreFoundation                 0x1b18ee0d0 __CFSafelyReallocate
5  Foundation                     0x1b1d4efa4 _NSMutableDataGrowBytes
6  Foundation                     0x1b1c9aaa8 -[NSConcreteMutableData appendBytes:length:]
7  ImageIO                        0x1b22089fc IIOImageWriteSession::putBytes(void const*, unsigned long)
8  ImageIO                        0x1b21df0e8 iioWriteCallback(void*, int, void*)
9  AppleJPEG                      0x1dbab69ec aj_ostream_write
10 AppleJPEG                      0x1dbad85dc aj_block_encode_ac
11 AppleJPEG                      0x1dbaa8990 aj_encode_buffers_baseline
12 AppleJPEG                      0x1dbaa8dc4 aj_encode_row
13 AppleJPEG                      0x1dbad9d84 applejpeg_encode_image_row
14 AppleJPEG                      0x1dbac0d24 recode_all
15 ImageIO                        0x1b228c998 IIOImageDestination::finalize()
16 ImageIO                        0x1b2291158 CGImageDestinationFinalize
17 PRJCT Prod                     0x104e1ad1c -[SDImageIOCoder encodedDataWithImage:format:options:] + 224 (SDImageIOCoder.m:224)
18 PRJCT Prod                     0x104e18130 -[SDImageCodersManager encodedDataWithImage:format:options:] + 127 (SDImageCodersManager.m:127)
19 PRJCT Prod                     0x104e10530 __71-[SDImageCache storeImage:imageData:forKey:toMemory:toDisk:completion:]_block_invoke + 197 (SDImageCache.m:197)
20 libdispatch.dylib              0x1b169d610 _dispatch_call_block_and_release
21 libdispatch.dylib              0x1b169e184 _dispatch_client_callout
22 libdispatch.dylib              0x1b167b73c _dispatch_lane_serial_drain$VARIANT$armv81
23 libdispatch.dylib              0x1b167c154 _dispatch_lane_invoke$VARIANT$armv81
24 libdispatch.dylib              0x1b168543c _dispatch_workloop_worker_thread
25 libsystem_pthread.dylib        0x1b16edfa4 _pthread_wqthread
26 libsystem_pthread.dylib        0x1b16f0ae0 start_wqthread

#0. com.apple.main-thread
0  libsystem_kernel.dylib         0x1b17a75f4 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b17a6a60 mach_msg + 72
2  CoreFoundation                 0x1b194e068 __CFRunLoopServiceMachPort + 216
3  CoreFoundation                 0x1b1949188 __CFRunLoopRun + 1444
4  CoreFoundation                 0x1b19488bc CFRunLoopRunSpecific + 464
5  GraphicsServices               0x1bb7b4328 GSEventRunModal + 104
6  UIKitCore                      0x1b59de6d4 UIApplicationMain + 1936
7  PRJCT Prod                     0x1044d10f8 main + 19 (ExperimentsServiceContext.swift:19)
8  libdyld.dylib                  0x1b17d3460 start + 4

#1. com.apple.uikit.eventfetch-thread
0  libsystem_kernel.dylib         0x1b17a75f4 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b17a6a60 mach_msg + 72
2  CoreFoundation                 0x1b194e068 __CFRunLoopServiceMachPort + 216
3  CoreFoundation                 0x1b1949188 __CFRunLoopRun + 1444
4  CoreFoundation                 0x1b19488bc CFRunLoopRunSpecific + 464
5  Foundation                     0x1b1c88994 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 228
6  Foundation                     0x1b1c88874 -[NSRunLoop(NSRunLoop) runUntilDate:] + 88
7  UIKitCore                      0x1b5a7649c -[UIEventFetcher threadMain] + 152
8  Foundation                     0x1b1db90b0 __NSThread__start__ + 848
9  libsystem_pthread.dylib        0x1b16ed1ec _pthread_start + 124
10 libsystem_pthread.dylib        0x1b16f0aec thread_start + 8

#2. com.twitter.crashlytics.ios.MachExceptionServer
0  libsystem_kernel.dylib         0x1b17a75f4 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b17a6a60 mach_msg + 72
2  PRJCT Prod                     0x104e8b9ec CLSMachExceptionServer + 180 (CLSMachException.c:180)
3  libsystem_pthread.dylib        0x1b16ed1ec _pthread_start + 124
4  libsystem_pthread.dylib        0x1b16f0aec thread_start + 8

#3. com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x1b17a75f4 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b17a6a60 mach_msg + 72
2  CoreFoundation                 0x1b194e068 __CFRunLoopServiceMachPort + 216
3  CoreFoundation                 0x1b1949188 __CFRunLoopRun + 1444
4  CoreFoundation                 0x1b19488bc CFRunLoopRunSpecific + 464
5  CFNetwork                      0x1b4bc3e68 (Missing)
6  Foundation                     0x1b1db90b0 __NSThread__start__ + 848
7  libsystem_pthread.dylib        0x1b16ed1ec _pthread_start + 124
8  libsystem_pthread.dylib        0x1b16f0aec thread_start + 8

#4. JavaScriptCore bmalloc scavenger
0  libsystem_kernel.dylib         0x1b17c8c8c __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x1b16ea4e0 _pthread_cond_wait$VARIANT$armv81 + 676
2  libc++.1.dylib                 0x1b181a7a8 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 24
3  JavaScriptCore                 0x1c05a63fc void std::__1::condition_variable_any::wait<std::__1::unique_lock<bmalloc::Mutex> >(std::__1::unique_lock<bmalloc::Mutex>&) + 108
4  JavaScriptCore                 0x1c05a9dcc bmalloc::Scavenger::threadRunLoop() + 200
5  JavaScriptCore                 0x1c05a9ab4 bmalloc::Scavenger::Scavenger(std::__1::lock_guard<bmalloc::Mutex>&) + 10
6  JavaScriptCore                 0x1c05aadc0 std::__1::__thread_specific_ptr<std::__1::__thread_struct>::set_pointer(std::__1::__thread_struct*) + 38
7  libsystem_pthread.dylib        0x1b16ed1ec _pthread_start + 124
8  libsystem_pthread.dylib        0x1b16f0aec thread_start + 8

#5. WebThread
0  libsystem_kernel.dylib         0x1b17a75f4 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b17a6a60 mach_msg + 72
2  CoreFoundation                 0x1b194e068 __CFRunLoopServiceMachPort + 216
3  CoreFoundation                 0x1b1949188 __CFRunLoopRun + 1444
4  CoreFoundation                 0x1b19488bc CFRunLoopRunSpecific + 464
5  WebCore                        0x1b9803100 RunWebThread(void*) + 612
6  libsystem_pthread.dylib        0x1b16ed1ec _pthread_start + 124
7  libsystem_pthread.dylib        0x1b16f0aec thread_start + 8

#6. Thread
0  libsystem_kernel.dylib         0x1b17c9a74 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1b16edff0 _pthread_wqthread + 352
2  libsystem_pthread.dylib        0x1b16f0ae0 start_wqthread + 8

#7. Crashed: com.twitter.crashlytics.ios.exception
0  PRJCT Prod                     0x104ea02a0 CLSProcessRecordAllThreads + 376 (CLSProcess.c:376)
1  PRJCT Prod                     0x104ea0688 CLSProcessRecordAllThreads + 407 (CLSProcess.c:407)
2  PRJCT Prod                     0x104e9038c CLSHandler + 26 (CLSHandler.m:26)
3  PRJCT Prod                     0x104e9e8b4 __CLSExceptionRecord_block_invoke + 198 (CLSException.mm:198)
4  libdispatch.dylib              0x1b169e184 _dispatch_client_callout + 16
5  libdispatch.dylib              0x1b1681dd8 _dispatch_lane_barrier_sync_invoke_and_complete + 56
6  PRJCT Prod                     0x104e9e35c CLSExceptionRecord + 205 (CLSException.mm:205)
7  PRJCT Prod                     0x104e9e190 CLSExceptionRecordNSException + 102 (CLSException.mm:102)
8  PRJCT Prod                     0x104e9ddb4 CLSTerminateHandler() + 258 (CLSException.mm:258)
9  libc++abi.dylib                0x1b179e304 std::__terminate(void (*)()) + 16
10 libc++abi.dylib                0x1b179dc58 __cxa_get_exception_ptr + 30
11 libc++abi.dylib                0x1b179dc18 __cxxabiv1::exception_cleanup_func(_Unwind_Reason_Code, _Unwind_Exception*) + 122
12 libobjc.A.dylib                0x1b16f91d0 _objc_exception_destructor(void*) + 354
13 CoreFoundation                 0x1b1a2c2b8 _CFPreferencesCopyManagedPreferencesContainerURL.cold.1 + 46
14 CoreFoundation                 0x1b1a24b54 __CFReallocationFailed + 120
15 CoreFoundation                 0x1b18ee0d0 __CFSafelyReallocate + 68
16 Foundation                     0x1b1d4efa4 _NSMutableDataGrowBytes + 344
17 Foundation                     0x1b1c9aaa8 -[NSConcreteMutableData appendBytes:length:] + 336
18 ImageIO                        0x1b22089fc IIOImageWriteSession::putBytes(void const*, unsigned long) + 184
19 ImageIO                        0x1b21df0e8 iioWriteCallback(void*, int, void*) + 36
20 AppleJPEG                      0x1dbab69ec aj_ostream_write + 392
21 AppleJPEG                      0x1dbad85dc aj_block_encode_ac + 1336
22 AppleJPEG                      0x1dbaa8990 aj_encode_buffers_baseline + 52
23 AppleJPEG                      0x1dbaa8dc4 aj_encode_row + 180
24 AppleJPEG                      0x1dbad9d84 applejpeg_encode_image_row + 580
25 AppleJPEG                      0x1dbac0d24 recode_all + 1300
26 ImageIO                        0x1b228c998 IIOImageDestination::finalize() + 5040
27 ImageIO                        0x1b2291158 CGImageDestinationFinalize + 124
28 PRJCT Prod                     0x104e1ad1c -[SDImageIOCoder encodedDataWithImage:format:options:] + 224 (SDImageIOCoder.m:224)
29 PRJCT Prod                     0x104e18130 -[SDImageCodersManager encodedDataWithImage:format:options:] + 127 (SDImageCodersManager.m:127)
30 PRJCT Prod                     0x104e10530 __71-[SDImageCache storeImage:imageData:forKey:toMemory:toDisk:completion:]_block_invoke + 197 (SDImageCache.m:197)
31 libdispatch.dylib              0x1b169d610 _dispatch_call_block_and_release + 24
32 libdispatch.dylib              0x1b169e184 _dispatch_client_callout + 16
33 libdispatch.dylib              0x1b167b73c _dispatch_lane_serial_drain$VARIANT$armv81 + 564
34 libdispatch.dylib              0x1b167c154 _dispatch_lane_invoke$VARIANT$armv81 + 400
35 libdispatch.dylib              0x1b168543c _dispatch_workloop_worker_thread + 576
36 libsystem_pthread.dylib        0x1b16edfa4 _pthread_wqthread + 276
37 libsystem_pthread.dylib        0x1b16f0ae0 start_wqthread + 8

#8. Thread
0  libsystem_kernel.dylib         0x1b17c9a74 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1b16edff0 _pthread_wqthread + 352
2  libsystem_pthread.dylib        0x1b16f0ae0 start_wqthread + 8

#9. Thread
0  libsystem_pthread.dylib        0x1b16f0ad8 start_wqthread + 190

#10. Thread
0  libsystem_kernel.dylib         0x1b17c9a74 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1b16edff0 _pthread_wqthread + 352
2  libsystem_pthread.dylib        0x1b16f0ae0 start_wqthread + 8

#11. Thread
0  libsystem_kernel.dylib         0x1b17c9a74 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1b16edff0 _pthread_wqthread + 352
2  libsystem_pthread.dylib        0x1b16f0ae0 start_wqthread + 8

#12. Thread
0  libsystem_pthread.dylib        0x1b16f0ad8 start_wqthread + 190
```
